ProcessLib/Process.cpp

```diff
@@ -58,56 +68,69 @@ void Process::initialize()
     finishNamedFunctionsInitialization();
 
     DBUG("Initialize boundary conditions.");
-    _boundary_conditions.addBCsForProcessVariables(
-        _process_variables, *_local_to_global_index_map, _integration_order);
-
-    for (int variable_id = 0;
-         variable_id < static_cast<int>(_process_variables.size());
-         ++variable_id)
+    for (std::size_t pcs_id = 0; pcs_id < _process_variables.size(); pcs_id++)
     {
-        ProcessVariable& pv = _process_variables[variable_id];
-        auto sts =
-            pv.createSourceTerms(*_local_to_global_index_map, variable_id,
-                                 _integration_order);
+        auto const& per_process_variables = _process_variables[pcs_id];
+        auto& per_process_BCs = _boundary_conditions[pcs_id];
+        std::vector<std::unique_ptr<NodalSourceTerm>> per_process_source_terms;
+        for (std::size_t variable_id = 0;
+             variable_id < per_process_variables.size();
+             variable_id++)
+        {
+            per_process_BCs.addBCsForProcessVariables(
+                per_process_variables, *_local_to_global_index_map,
+                _integration_order);
+
+            ProcessVariable& pv = per_process_variables[variable_id];
```

Comment:
This is good candidate for the for-range loop
```c++
for (auto& pv : per_process_variables) { ... }
```

Reply:
Took it.